cmd/gpu_nfdhook/labeler.go

```diff
@@ -245,40 +282,125 @@ scanning:
 	}
 }
 
+// this returns pci groups label value, groups separated by "_", gpus separated by ".".
+// Example for two groups with 4 gpus: "0.1.2.3_4.5.6.7".
+func (l *labeler) createPCIGroupLabel(gpuNumList []string) string {
+	pciGroups := map[string][]string{}
+
+	pciGroupLevel := getEnvVarNumber(pciGroupingEnv)
+	if pciGroupLevel == 0 {
+		return ""
+	}
+
+	for _, gpuNum := range gpuNumList {
+		symLinkTarget, err := filepath.EvalSymlinks(path.Join(l.sysfsDRMDir, "card"+gpuNum))
+
+		if err == nil {
+			if pathPart := getPCIPathParts(pciGroupLevel, symLinkTarget); pathPart != "" {
+				pciGroups[pathPart] = append(pciGroups[pathPart], gpuNum)
+			}
+		}
+	}
+
+	labelValue := ""
+	separator := ""
+
+	// process in stable order by sorting
+	keys := []string{}
+	for key := range pciGroups {
+		keys = append(keys, key)
+	}
+
+	sort.Strings(keys)
+
+	for _, key := range keys {
+		labelValue = labelValue + separator + strings.Join(pciGroups[key], ".")
+		separator = "_"
+	}
+
+	return labelValue
+}
+
+// split returns the given string cut to chunks of size up to last separator below maxLength size.
+// Separator is not returned in the beginning or end of chunks.
+func split(str string, maxLength uint, separator string) []string {
+	remainingString := strings.TrimSuffix(str, separator)
+	lastSeparatorIndex := 0
+	results := []string{}
+
+	for lastSeparatorIndex >= 0 {
+		remainingString = strings.TrimPrefix(remainingString[lastSeparatorIndex:], separator)
+
+		if uint(len(remainingString)) <= maxLength {
+			results = append(results, remainingString)
+			return results
+		}
```

Comment:
`split("1.2.3.4_5.6.7_8.9", 6, "_")` should return an empty slice and for `split("1.2.3_4.5.6.7_8.9", 6, "_")` we expect `["1.2.3"]`?

Reply:
I added more tests to clarify and a comment about oversized strings without separators. For us the use-case is to limit labels under k8s label value length limit of 63, and we have densely separators in our inputs.

Reply:
In hindsight I figured you are right and simplified the whole thing. The users just need to concatenate, and check for extra label values if the current label is 63 chars long.